### PR TITLE
Improve  cli/transpilation docs

### PR DESCRIPTION
**Pull request by Izhaki**

### CODE REVIEW

**Comment by Izhaki** [comment]
Related: #1100 #1161

File: docs/cli.md [hunk 1]
```diff
@@ -114,9 +114,54 @@ In order to store and reuse commonly used CLI options, you can add a `cucumber.j
...
+
+or [CoffeeScript](https://www.npmjs.com/package/coffee-script):
+
+```
+--require-module coffee-script/register --require 'features/**/*.coffee'
+```
+
+### Extra configuration
+
+Sometimes the required module (say `@babel/register`) needs extra configuration. In such cases create a script (say, `cucumber.config.js`):
```
Comment (charlierudolph) [inline docs/cli.md]: Thoughts on the sample file being `babel_register.js`? Its not cucumber specific and could actually be used if you have mocha unit tests.

Comment (Izhaki) [reply docs/cli.md]: Naming things!

I completely see where you are coming from, but this is SOOOOOO context dependent!

One things we have done in the past is `chai.use(chaiAsPromised);` in such script. But I guess that's for Mocha as well (but `babel` in the name won't make sense).

Possibly `tests.setup.js` is another candidate... doubt anything will come out as an ultimate winner here...

Anyhow, changed to `babel.register.js`.

(Don't mind changing to `tests.setup.js` though).

Views?

Comment (charlierudolph) [reply docs/cli.md]: Either `babel.register.js` or `tests.setup.js` work for me.

File: docs/cli.md [hunk 2]
```diff
@@ -114,9 +114,54 @@ In order to store and reuse commonly used CLI options, you can add a `cucumber.j
...
+require('@babel/register')({
+  rootMode: 'upward',
+  ignore: ['node_modules'],
+});
+```
+
+And then require it using the `--require` option:
+
+```
+--require cucumber.config.js --require 'features/**/*.js'
```
Comment (charlierudolph) [inline docs/cli.md]: This needs to be updated now to `babel.register.js`

**Comment by Izhaki** [comment]
Renamed to `tests.setup.js`.

Final look to ensure no typos/issues?

**Comment by charlierudolph** [comment]
Thanks for this!

**Pull request merged**